include/Yash.h

```diff
@@ -446,27 +373,18 @@ class Yash {
     static constexpr const char* s_moveCursorForward = "\033[1C";
     static constexpr const char* s_moveCursorBackward = "\033[1D";
     static constexpr const char* s_commandDelimiter = " ";
-    std::map<std::string, Function> m_functions;
-    std::vector<std::string> m_commands;
-    std::vector<std::string>::const_iterator m_commandsIndex;
-    size_t m_pos{0};
+    static constexpr std::array<std::string_view, 9> s_ctrlCharacters { { { "A" }, { "B" }, { "C" }, { "D" }, { "1~" }, { "3~" }, { "4~" }, { "1;5C" }, { "1;5D" } } };
+
+    CtrlState m_ctrlState { CtrlState::None };
+    const std::array<Command, TCommandArraySize>& m_commands;
+    std::function<void(const char*)> m_printFunction;
+    std::vector<std::string> m_commandHistory;
```

Comment:
I suggest we use `std::list` instead of `std::vector`, because vector requires a continues memory area to place the command history, and since the size is not reserved, the placement of the vector can potentially jump address space each time it needs to expand the memory area if there is not enough free memory to expand.

Worst case is an out of memory fault when the vector is unable to expand.

Reply:
Yes. I actually wanted to do this in an update but left it out here (already too many changes - sorry)

Reply:
Sure, ill wrap up a branch to change it.